Week2/homework/js-exercises/5-who-wants-a-drink.js

```diff
@@ -0,0 +1,26 @@
+{
+    'use strict';
+
+    const drinkTypes = ['cola', 'lemonade', 'water'];
+    let drinkTray = [];
+
+    for (let i = 0; i < drinkTypes.length; ++i) {
+      let currentDrinkType = drinkTypes[i];
+      let drinkCount = 0;
+
+      do {
+        // check that there are not more than 2 of a drinkType already in drinkTray
+        // if true, then push
+        for (let drink of drinkTray) {
+          if (drink === currentDrinkType) {
+            drinkCount++;
+          };
+        }
+        if (drinkCount < 2) {
+          drinkTray.push(currentDrinkType);
+        }
+      } while (drinkCount < 2 && drinkTray.length < 5);
+    }
```

Comment:
Ah. I see what you tried to do here but don't you think the logic seems a lot complex with 2 for loops, 1 do-while loop and 2 if conditions. A simpler logic like below should also work fine, right?

```
for (i = 0; i < 5; i++) {
  drinkTray.push(drinkTypes[i % drinkTypes.length]);
}
```